py/dml/messages.py

```diff
@@ -633,6 +633,24 @@ def __init__(self, fun, ftype = None):
             ftype = fun.ctype()
         DMLError.__init__(self, fun, fun, ftype)
 
+class EAPPLYMETH(DMLError):
+    """
+    Calls to methods that may throw or multiple output parameters cannot be
+    used as arbitrary expressions. In DML 1.2, any such method must be called
+    via the `call` or `inline` statements, and in DML 1.4 any such method must
+    be called either as a standalone statement, as the RHS of an assignment, or
+    as the argument of a `return` statement.
+    """
+    fmt = "call to method '%s' in unsupported context\n%s"
+    def __init__(self, site, fun):
+        if site.dml_version() == (1, 2):
+            suggestion = ("use the 'call' or 'inline' statements to call "
+                          + "this method")
+        else:
+            suggestion = ("perform this method call either as a standalone "
```

Comment:
For multi-line, please add site to the second line. The current convention is:
```
    def log(self):
        DMLError.log(self)
        self.print_site_message(self.site, "...")
```
If you find this too verbose, then I wouldn't mind polishing the idiom.

Reply:
That's not at all the convention. Multi-line error messages use `\n` all over the place, `self.print_site_message` is typically used only when there are multiple sites; in fact, there is only one case where it's used with `self.site`, namely `EINC`.

Reply:
ok, sorry for the noise